### PR TITLE
add weak memcpy et al (take 2)

### PR DESCRIPTION
**Pull request by japaric**
cc @Amanieu

### CODE REVIEW

**Comment by japaric** [comment]
Splitting rlibc into its own crate yields correct `__aeabi_*` intrinsics 

```
00000000 <__aeabi_memcpy>:
   0:   eafffffe        b       0 <memcpy>
```

But using this crate still generates a recursive bomb.

**Comment by Amanieu** [comment]
So is the problem inside the rlibc crate? Is the `memcpy` function calling itself?

**Comment by japaric** [comment]
The rlibc disassembly loosk correct.

Interestingly, disabling LTO on this branch doesn't fix the infinite recursion.

**Comment by Amanieu** [comment]
`memcmp` should be added as well since it is called by libcore.

File: src/lib.rs [hunk 1]
```diff
@@ -9,6 +9,8 @@
 #[cfg(test)]
 extern crate core;
 
+extern crate rlibc;
```
Comment (Amanieu) [inline src/lib.rs]: I don't think an `extern crate` is necessary here since you're getting the symbols using `extern "C"`.

**Pull request closed without merging**